src/ui/App.tsx

```diff
@@ -0,0 +1,596 @@
+import React, { useState, useEffect } from 'react';
+import { createRoot } from 'react-dom/client';
+import ReactMarkdown from 'react-markdown';
+import { SemVer } from 'semver';
+
+interface Package {
+  name: string;
+  version: string;
+  location: string;
+}
+
+type ReleaseType =
+  | 'major'
+  | 'minor'
+  | 'patch'
+  | 'intentionally-skip'
+  | 'custom'
+  | string;
+
+function App() {
```

Comment:
I'd try breaking this component up, since there is a lot going on, and it's a bit hard to follow right now.

Reply:
yes currently doing this